app/jobs/complete_backorder_job.rb

```diff
@@ -17,7 +30,20 @@ def perform(user, order_id)
   # Our local stock can increase when users cancel their orders.
   # But stock levels could also have been adjusted manually. So we review all
   # quantities before finalising the order.
-  def adjust_quantities(order)
-    # TODO
+  def adjust_quantities(user, order, variants)
+    broker = FdcOfferBroker.new(BackorderJob.load_catalog(user))
+
+    order.lines.each do |line|
+      wholesale_product_id = line.offer.offeredItem.semanticId
+      transformation = broker.wholesale_to_retail(wholesale_product_id)
+      linked_variant = variants.linked_to(transformation.retail_product_id)
+
+      # Note that a division of integers dismisses the remainder, like `floor`:
+      wholesale_items_contained_in_stock = linked_variant.on_hand / transformation.factor
+      line.quantity = line.quantity.to_i - wholesale_items_contained_in_stock
+
+      retail_stock_changes = wholesale_items_contained_in_stock * transformation.factor
+      linked_variant.on_hand -= retail_stock_changes
```

Comment:
I have to confess my brain didn't quite make it to the end of the PR before turning into mush. 
It seems that we're adjusting the backorders relative to what they were before. Would it not be more robust to look at all current retail orders and re-calculate the backorder requirement from scratch? 
Hmm.. no we can't because we don't know how much of our "on_hand" stock was from a backorder, unless we look at the backorder.

I can't work out if that's a problem or not 🤯

Reply:
Yeah, it's a tricky one. I would also favour recalculating from scratch, but the use of OFN stock is so much more helpful/flexible for this one (we've already encountered an edge case for one of our pilot hubs that will need to utilise that)...

..so long as everything always works, I think we're good! 🫣